aea/cli/fetch.py

```diff
@@ -159,29 +160,26 @@ def fetch_agent_ipfs(
 
     if target_dir is None:
         target_dir = Path(ctx.cwd).absolute()
+        target_dir = target_dir / (alias or public_id.name)
+
+    target_dir = cast(Path, target_dir)
+    if target_dir.exists():
+        raise click.ClickException(f"Package already exists at {target_dir}")
 
     ipfs_tool = IPFSTool(get_ipfs_node_multiaddr())
 
     try:
-        agent_path = ipfs_tool.download(public_id.hash, str(target_dir))
-        ctx.clean_paths.append(agent_path)
-        if alias is not None:
-            target_dir = cast(Path, target_dir) / alias
-            if target_dir.exists():
-                raise click.ClickException(f"{alias} already exists at {target_dir}")
```

Comment:
This piece of code was causing this behaviour https://github.com/valory-xyz/open-autonomy/issues/1886. Since we're moving the downloaded package to the intended directory and then moving it the alias directory the IPFS tool was raising the error if package existed even though the alias was provided.

Say you downloaded a package names `hello_world` and you try to download the package with different alias it wouldn't have worked since we move the package to alias dir after downloading.